subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/PathUtil.java

```diff
@@ -161,10 +304,10 @@ private static int lastIndexOfSeparator(String absolutePath) {
     }
 
     /**
-     * This uses an optimized version of {@link String#regionMatches(int, String, int, int)}
-     * which does not check for negative indices or integer overflow.
+     * Checks whether the filePath starting at the offset has the given prefix.
      */
-    public static boolean isChildOfOrThis(String filePath, int offset, String prefix) {
+    public static boolean isChildOfOrThis(String prefix, String filePath, int offset, CaseSensitivity caseSensitivity) {
```

Comment:
Can we capture "child-of or this" in a single term somehow? It is used in several places, and it's really confusing.